### ISSUE TITLE
BulkData Export checks that the Resource Types exported are valid based on the fhir-model and not configurations

### ISSUE THREAD

**Issue by prb112** [IBM/FHIR#3083]
**Describe the bug**

BulkData Export checks that the Resource Types exported are valid based on the fhir-model. 
In specific configurations users may choose to deploy only a subset of ResourceTypes

Bulk Data should verify the resources are supported in the configuration or in the custom compartment.

```
            if (ExportType.PATIENT.equals(exportType) || ExportType.GROUP.equals(exportType)) {
                if (types != null && !types.isEmpty()) {
                    export.checkExportPatientResourceTypes(types);
                } else {
                    types = export.addDefaultsForPatientCompartment();
                }
            } else if ((ExportType.SYSTEM.equals(exportType) )
                            && (types == null || types.isEmpty())) {
                types = export.defaultResourceTypes();
            }
```

```
public class BulkDataExportUtil {
    private static Set<String> RESOURCE_TYPES = ModelSupport.getResourceTypes(false).stream()
                                                    .map(Class::getSimpleName)
                                                    .collect(Collectors.toSet());
```

**Environment**
Which version of IBM FHIR Server? main

**To Reproduce**

**Expected behavior**

**Additional context**

**Comment by lmsurpre** [comment]
Same sort of issue as #2963 ...would be nice to have a utility we can use which abstracts the config and just gives us a list of which resource types are enabled.

**Comment by lmsurpre** [comment]
when #3355 gets merged, this is expected to work in the following cases:
1. `/fhirServer/resources` config where `open=false` and one or more resource types are listed
2. no `/fhirServer/resources` config (i.e. everything supported)

It is not expected to work in the case of `open=true` with one or more resource types listed.
The fix for that is in the `r4b-rebased` branch.

@prb112 any thoughts on whether to keep this one open (probably mark it blocked by that other one and unassign you) vs opening a separate issue to track that?

**Comment by prb112** [comment]
How about we target the r4b-rebased branch for this one? Thoughts?
I could also do a custom thing where we check open=false.  

I'd like to close it out as complete, rather than leave it incomplete

**Comment by lmsurpre** [comment]
To me, its looking more and more likely like we'll want to cut a release prior to R4B being ready, so I have a slight preference for coding up the logic to handle this case in the current PR (if its not too much trouble)

**Comment by prb112** [comment]
QA: Configure a open: false for resource then run a request for an invalid Resource

**Comment by d0roppe** [comment]
Setting the value to false, and then doing a $export of the whole db, caused the request to fail, and no data was exported.  Would have expected the $export to work on the resources mentioned in the fhir-server-config.json fhirServer/resoures section

**Comment by d0roppe** [comment]
Still not working with open=false and $export

**Comment by prb112** [comment]
fhir-server-config.json requires id and _lastUpdated + PR I just appended a fix to

``` json
{
    "fhirServer": {
        "resources": {
            "open": false,
            "CarePlan": {
                "searchIncludes": [
                    "CarePlan:date",
                    "CarePlan:performer"
                ],
                "searchParameters": {
                    "_id": "http://hl7.org/fhir/SearchParameter/Resource-id",
                    "_lastUpdated": "http://hl7.org/fhir/SearchParameter/Resource-lastUpdated",
                    "category": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careplan-category",
                    "date": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careplan-date",
                    "encounter": "http://hl7.org/fhir/SearchParameter/CarePlan-encounter",
                    "patient": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careplan-patient",
                    "performer": "http://hl7.org/fhir/SearchParameter/CarePlan-performer",
                    "status": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careplan-status"
                }
            },
            "CareTeam": {
                "searchIncludes": [
                    "CareTeam:date",
                    "CareTeam:participant"
                ],
                "searchParameters": {
                    "_id": "http://hl7.org/fhir/SearchParameter/Resource-id",
                    "_lastUpdated": "http://hl7.org/fhir/SearchParameter/Resource-lastUpdated",
                    "date": "http://hl7.org/fhir/SearchParameter/clinical-date",
                    "encounter": "http://hl7.org/fhir/SearchParameter/CareTeam-encounter",
                    "participant": "http://hl7.org/fhir/SearchParameter/CareTeam-participant",
                    "patient": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careteam-patient",
                    "status": "http://hl7.org/fhir/us/core/SearchParameter/us-core-careteam-status"
                }
            },            
            "Location": {
                "searchIncludes": [
                ],
                "searchParameters": {
                    "_id": "http://hl7.org/fhir/SearchParameter/Resource-id",
                    "_lastUpdated": "http://hl7.org/fhir/SearchParameter/Resource-lastUpdated",
                    "address": "http://hl7.org/fhir/us/core/SearchParameter/us-core-location-address",
                    "address-city": "http://hl7.org/fhir/us/core/SearchParameter/us-core-location-address-city",
                    "address-postalcode": "http://hl7.org/fhir/us/core/SearchParameter/us-core-location-address-postalcode",
                    "address-state": "http://hl7.org/fhir/us/core/SearchParameter/us-core-location-address-state",
                    "name": "http://hl7.org/fhir/us/core/SearchParameter/us-core-location-name"
                }
            },
            "Organization": {
                "searchIncludes": [
                ],
                "searchParameters": {
                    "_id": "http://hl7.org/fhir/SearchParameter/Resource-id",
                    "_lastUpdated": "http://hl7.org/fhir/SearchParameter/Resource-lastUpdated",
                    "address": "http://hl7.org/fhir/us/core/SearchParameter/us-core-organization-address",
                    "name": "http://hl7.org/fhir/us/core/SearchParameter/us-core-organization-name",
                    "endpoint": "http://hl7.org/fhir/us/davinci-pdex-plan-net/SearchParameter/organization-endpoint"
                }
            },
            "Patient": {
                "searchIncludes": [
                    "Patient:general-practitioner",
                    "Patient:organization"
                ],
                "searchParameters": {
                    "_id": "http://hl7.org/fhir/SearchParameter/Resource-id",
                    "_lastUpdated": "http://hl7.org/fhir/SearchParameter/Resource-lastUpdated",
                    "family": "http://hl7.org/fhir/SearchParameter/individual-family",
                    "favorite-color": "http://ibm.com/fhir/SearchParameter/Patient-favorite-color",
                    "favorite-number": "http://ibm.com/fhir/SearchParameter/Patient-favorite-number",
                    "favorite-code": "http://ibm.com/fhir/SearchParameter/Patient-favorite-code",
                    "favorite-uri": "http://ibm.com/fhir/SearchParameter/Patient-favorite-uri",
                    "favorite-quantity": "http://ibm.com/fhir/SearchParameter/Patient-favorite-quantity",
                    "favorite-date": "http://ibm.com/fhir/SearchParameter/Patient-favorite-date",
                    "gender": "http://hl7.org/fhir/SearchParameter/individual-gender",
                    "general-practitioner": "http://hl7.org/fhir/SearchParameter/Patient-general-practitioner",
                    "identifier": "http://hl7.org/fhir/SearchParameter/Patient-identifier",
                    "organization": "http://hl7.org/fhir/SearchParameter/Patient-organization"
                }
            }
        }
}
```

**Comment by prb112** [comment]
QA: @d0roppe note the changes to the fhir-server-config.json
It should be consistent with group/patient/system exports

**Comment by d0roppe** [comment]
Tested with the new values.  Working as expected
